rts/Game/UI/GuiHandler.cpp

```diff
@@ -2345,7 +2352,9 @@ Command CGuiHandler::GetCommand(int mouseX, int mouseY, int buttonHint, bool pre
 			if (mouse->buttons[button].movement <= mouse->dragCircleCommandThreshold) {
 				const CUnit* unit = nullptr;
 				const CFeature* feature = nullptr;
-				const float dist2 = TraceRay::GuiTraceRay(cameraPos, mouseDir, camera->GetFarPlaneDist() * 1.4f, NULL, unit, feature, true);
+				const float viewRange = camera->GetFarPlaneDist() * 1.4f;
+				const float3 rayOrigin = camera->NearTheaterIntersection(mouseDir, viewRange);
+				const float dist2 = TraceRay::GuiTraceRay(rayOrigin, mouseDir, viewRange, NULL, unit, feature, true);
```

Comment:
Nit: perhaps a good idea to use the modern null while we're touching the line
```suggestion
				const float dist2 = TraceRay::GuiTraceRay(rayOrigin, mouseDir, viewRange, nullptr, unit, feature, true);
```

Reply:
sure

Reply:
done